src/polygon.jl

```diff
@@ -68,19 +68,62 @@ end
 
 
 """
-    get(code)
-Returns the MULTIPOLYGON data for the requested country.\n
-Input: ISO 3166 Alpha 3 Country Code
+    get(country, levels...)
+Returns the MULTIPOLYGON data for the requested region.\n
+Input: ISO3 Country Code, and further subdivisions\n
+
+Examples:\n
+1. `get("IND")` - Returns polygon of India
+2. `get("IND", "Uttar Pradesh")` - Returns polygon of the state Uttar Pradesh
+3. `get("IND", "Uttar Pradesh", "Lucknow")` - Returns polygon of district Lucknow\n
 """
-function get(code)
+function get(country, levels...) 
     # only uppercase country codes are accepted
-    isvalidcode(code) || throw(ArgumentError("please provide standard ISO 3166 Alpha 3 country codes"))
+    isvalidcode(country) || throw(ArgumentError("please provide standard ISO 3 country codes"))
 
-    dataid = "GADM_$code"
+    dataid = "GADM_$country"
 
     resource_data_path = download(dataid)
 
     dataset = extractdataset(resource_data_path)
 
-    extractgeometry(dataset)
+    # the number of varargs in levels is the required level for GADM dataset
+    required_level = length(levels)
+
+    # get layer of the required level
+    level = getlevel(dataset, required_level)
+
+    # get country level
+    if required_level == 0
+        return ArchGDAL.getfeature(level, 1) do feature
+            ArchGDAL.getgeom(feature)
+        end
+    end
+
+    nfeatures = ArchGDAL.nfeature(level)
+
+    # indexes of fields with entity full names
+    # name of i level entity is at index at indexes[i+1]
+    # Eg. Name of state (level 1) "Uttar Pradesh" will be found at index name_indexes[2], i.e. 3rd field
+    name_indexes = [0, 3, 6]
```

Comment:
Could you improve this comment? I couldn't grasp it. Maybe draw a tree or some data structure in the comment to illustrate what is happening.